orangecanvas/scheme/scheme.py

```diff
@@ -659,10 +659,8 @@ def weight(out_c, in_c):
             else:
                 # Does the connection type check (can only ever be False for
                 # dynamic signals)
-                type_checks, _ = _classify_connection(out_c, in_c)
-                # Dynamic signals that require runtime instance type check
-                # are considered last.
-                check = [type_checks,
+                strict, dynamic = _classify_connection(out_c, in_c)
+                check = [strict or dynamic,
```

Comment:
`strict or dynamic` will always be `True` so the entire 'check' entry can be removed. I.e.
```python
check = [
    in_c not in already_connected_sinks,
    bool(in_c.default),
    bool(out_c.default)
]
```
Also remove the preceding comment `# Does the connection type check (can only ever be False for dynamic signals)` since it no longer applies.